system/lib/gl/gl.c

```diff
@@ -2397,6 +2403,10 @@ for line in open('a').readlines():
   if (!strcmp(name, "glFramebufferTexture3D")) return emscripten_glFramebufferTexture3D;
   if (!strcmp(name, "glTexImage2D")) return emscripten_glTexImage2D;
   if (!strcmp(name, "glGetShaderPrecisionFormat")) return emscripten_glGetShaderPrecisionFormat;
+  if (!strcmp(name, "glDispatchCompute")) return emscripten_glDispatchCompute;
+  if (!strcmp(name, "glBindImageTexture")) return emscripten_glBindImageTexture;
+  if (!strcmp(name, "glMemoryBarrier")) return emscripten_glMemoryBarrier;
+  if (!strcmp(name, "glMemoryBarrierByRegion")) return emscripten_glMemoryBarrierByRegion;
```

Comment:
This also regresses code size, in a particularly tricky way. For this we should migrate to codegenning this differently per API level.

Reply:
FYI, many ES 3.1 APIs are still not added here, so the code size is expected to be increased further. Can we leave codegenning for future improvement?

Reply:
Is the regression from adding all these strings? Or is it somehow specific to glMemoryBarrierByRegion? Can this entire block just be inside of an ifdef?

Reply:
The think the regression is that these lines keep alive each of those functions whose address they return. So the compiler can't dce them out, they are all used.

We do build this file separately for when GL emulation is present. We could build separate versions for the different GL levels, to reduce the problem. That code is in `tools/system_libraries.py`.

Reply:
+1, sounds like a reasonable fix to me. FYI @hujiajie: In `tools/system_libs.py`, I'm looking at `create_gl`. It does `flags += legacy_gl_emulation_flags(libname)`. I think the suggestion is to add another `flags += webgl2_compute_flags()` similar to that.

Reply:
IIUC, the idea has already been implemented in https://github.com/emscripten-core/emscripten/commit/ef230f55772b7a2b2d4c6a29b8f1cde915bd8e3f. I'm rebasing on top of that.